ntbs-service-unit-tests/Services/EnhancedSurveillanceAlertsServiceTest.cs

```diff
@@ -74,9 +74,31 @@ public EnhancedSurveillanceAlertsServiceTest()
         }
 
         [Theory]
-        [InlineData("M. bovis", true, false)]
-        [InlineData("Non M. bovis", false, true)]
+        [InlineData("M. bovis", null, null, null, null, true, false)]
+        [InlineData("M. bovis", null, null, null, true, true, false)]
+        [InlineData("M. bovis", null, null, true, null, true, false)]
+        [InlineData("M. bovis", null, true, null, null, true, false)]
+        [InlineData("M. bovis", true, null, null, null, true, false)]
+        [InlineData("M. bovis", false, false, false, null, true, false)]
+        [InlineData("M. bovis", false, false, null, false, true, false)]
+        [InlineData("M. bovis", false, null, false, false, true, false)]
+        [InlineData("M. bovis", null, false, false, false, true, false)]
+        [InlineData("M. bovis", false, false, false, false, false, true)]
+        [InlineData("Non M. bovis", null, null, null, null, false, true)]
+        [InlineData("Non M. bovis", null, null, null, true, false, true)]
+        [InlineData("Non M. bovis", null, null, true, null, false, true)]
+        [InlineData("Non M. bovis", null, true, null, null, false, true)]
+        [InlineData("Non M. bovis", true, null, null, null, false, true)]
+        [InlineData("Non M. bovis", false, false, false, null, false, true)]
+        [InlineData("Non M. bovis", false, false, null, false, false, true)]
+        [InlineData("Non M. bovis", false, null, false, false, false, true)]
+        [InlineData("Non M. bovis", null, false, false, false, false, true)]
+        [InlineData("Non M. bovis", false, false, false, false, false, true)]
```

Comment:
Choosing which tests to run in a case like this is a delicate question. Obviously I wouldn't expect you to run all 162 possibilities!

In this case, I think that it would be good to add at least one more case where the species is `M. bovis` and the questionnaire has been filled out - maybe 2... one with all `true` values and one with a mix. I think we could probably get rid of most of the `Non M. bovis` cases, as I don't think they add much value. Perhaps there we could have one with an empty questionnaire, one partially filled out and one fully filled out.

Reply:
Yeah I agree, those seem to cover the cases a bit better